src/hdbscan.jl

```diff
@@ -0,0 +1,303 @@
+# HDBSCAN Graph
+# edge[i] is a list of edges adjacent to the i-th vertex
+# the second element of HDBSCANEdge is the mutual reachability distance.
+HDBSCANEdge = Tuple{Int, Float64}
+struct HDBSCANGraph
+    edges::Vector{Vector{HDBSCANEdge}}
+    HDBSCANGraph(nv::Integer) = new([HDBSCANEdge[] for _ in 1 : nv])
+end
+
+function add_edge!(G::HDBSCANGraph, v1::Integer, v2::Integer, dist::Number)
+    push!(G.edges[v1], (v2, dist))
+    push!(G.edges[v2], (v1, dist))
+end
+
+Base.getindex(G::HDBSCANGraph, i::Int) = G.edges[i]
+
+struct MSTEdge
+    v1::Integer
+    v2::Integer
+    dist::Number
+end
+expand(edge::MSTEdge) = (edge.v1, edge.v2, edge.dist)
+Base.isless(edge1::MSTEdge, edge2::MSTEdge) = edge1.dist < edge2.dist
+
+"""
+    HdbscanCluster
+A cluster generated by the [hdbscan](@ref) method, part of [HdbscanResult](@ref).
```

Comment:
```suggestion
    HdbscanCluster

A cluster generated by the [hdbscan](@ref) method, part of [HdbscanResult](@ref).
```

@MommaWatasu Please make sure that in all your docstrings there is an empty line between the declaration of the structure or a method and its description. That's the [docstring requirement](https://docs.julialang.org/en/v1/manual/documentation/#Writing-Documentation).